content/Documentation Guidelines - Sebastián Caraballo.md

```diff
@@ -0,0 +1,113 @@
+## **Why Documentation Matters**
+
+![main image](images/why_documenting_matters.jpg)
+
+### **Introduction**
+
+Here at Rootstrap, we help our clients reach their business goals by developing high-quality, performant & scalable software solutions. To achieve this, our team continuously focuses on improving all aspects of our processes through several initiatives across all company departments.
+
+In this article, I'll go deep on why documentation is vital for the software development cycle.
+
+### **Why should we document?**
+
+The reader may be wondering about the benefits of documenting. Let me tell you a brief story. In March 2020, we started working with a client on a very exciting project. However, the whole team switched to remote work due to COVID-19. So we had to focus on how to keep a fluent communication among frontend & backend developers. In order to do this, the backend team started to document the API features and the available deliverables right away with different tools. Doing this helped to sustain velocity, reduce consultation time between devs, and for everyone to be aware of each others work.
+
+Many of the features require backend work at first so that frontenders integrate their mocked code with the API afterwards. This now implies that backenders not only have to code and test, but also add documentation on each new feature release. This means that when frontend starts working with any API endpoint, they already have a detailed description of returned objects, whether authentication is needed or not, and error messages that may arise based on different use case scenarios.
```

Comment:
```suggestion
Many of the features require backend work to be done first so that frontenders can integrate their mocked code with the API afterwards. This now implies that backenders not only have to code and test, but also add documentation on each new released feature. By the time a frontender starts working with any API endpoint, they already have now a detailed description of returned objects, whether authentication is needed or not, and error messages that may arise based on different use case scenarios.
```